examples/llama.py

```diff
@@ -142,15 +142,8 @@ def __call__(self, tokens:Tensor, start_pos:int):
 
     # get only the part we are using. making it contiguous avoids more kernel calls
     freqs_cis = self.freqs_cis[:, start_pos:start_pos+seqlen].contiguous().realize()
-    if seqlen > 1:
-      mask = np.full((1, 1, seqlen, start_pos + seqlen), float("-inf"), dtype=np.float32)
-      mask = np.triu(mask, k=start_pos + 1)  # TODO: this is hard to do in tinygrad
-      mask = Tensor(mask)
-    else:
-      mask = None
-    # mask = Tensor.full((1, 1, seqlen, start_pos + seqlen), float("-inf"), dtype=dtypes.float32).triu(start_pos+1) if seqlen > 1 else None #TODO: Pending(#942)
+    mask = Tensor.full((1, 1, seqlen, start_pos + seqlen), float("-inf"), dtype=dtypes.float32).triu(start_pos+1).realize() if seqlen > 1 else None
     for layer in self.layers:
-      h.realize()  # TODO: why do i need this?
```

Comment:
Are you sure that this realize isn't required anymore?
I wonder if this was to prevent kernel compilation issues when there are lots of buffers.

Reply:
Probably that was the case. I tested on METAL and CPU and things seemed fine.